stag-library/src/main/java/com/vimeo/stag/KnownTypeAdapters.java

```diff
@@ -115,7 +115,7 @@ public void write(JsonWriter out, Short value) throws IOException {
         @Override
         public Integer read(JsonReader in) throws IOException {
             try {
-                return in.nextInt();
+                return Integer.valueOf(in.nextInt());
```

Comment:
Okay, yes, the reason for using `Integer.valueOf` makes sense because caching the objects. However, the whole point of the warning is because the compiler automatically applies the caching post java 1.5. Calling `Integer integer = Integer.valueOf(1)` will produce the same java bytecode as `Integer integer = 1`, and for this reason the warning exists.

I've written a unit test demonstrating that this is the case. Here we test object reference equality rather than using the `equals` method to ensure that all objects are pointing to the same memory location. If you run this test, you should see that when in both cases, the integer is <= 127, the object is cached by the `Integer` class. When it's greater, it allocates a new object. I've tested this using JUnit, on the Emulator, and on my Nexus 5X and verified the behavior is the same everywhere.

```java
public class AutoBoxingTest {

    @Test
    public void testBoxing_equal(){
        Object integer1 = Integer.valueOf(1);
        Object integer2 = Integer.valueOf(1);

        Assert.assertTrue(integer1 == integer2);
    }

    @Test
    public void testBoxing_notEqual(){
        Object integer1 = Integer.valueOf(Integer.MAX_VALUE);
        Object integer2 = Integer.valueOf(Integer.MAX_VALUE);

        Assert.assertFalse(integer1 == integer2);
    }

    @Test
    public void testNoBoxing_notEqual(){
        Object integer1 = Integer.MAX_VALUE;
        Object integer2 = Integer.MAX_VALUE;

        Assert.assertFalse(integer1 == integer2);
    }

    @Test
    public void testNoBoxing_equal(){
        Object integer1 = 127;
        Object integer2 = 127;

        Assert.assertTrue(integer1 == integer2);
    }

}
```

For this reason I am proposing the removal of the autoboxing, as it makes the code unnecessarily complicated.